MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/ProductServiceDefault.java

```diff
@@ -166,60 +161,89 @@ public CancellableTask getProducts(final List<Long> productIds, final Callback<L
+    public Observable<List<Collection>> getCollections(final int page) {
+        if (page < 1) {
+            throw new IllegalArgumentException("page is a 1-based index, value cannot be less than 1");
+        }
+
+        // All collection responses from the server are wrapped in a CollectionListings object which contains and array of collections
+        // For this call, we will clamp the size of the collection array returned to the page size
+        return retrofitService
+            .getCollectionPage(appId, page, pageSize)
+            .retryWhen(networkRetryPolicyProvider.provide())
+            .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>())
+            .compose(new UnwrapRetrofitBodyTransformer<CollectionListings, List<Collection>>())
+            .onErrorResumeNext(new BuyClientExceptionHandler<List<Collection>>())
+            .observeOn(callbackScheduler);
     }
 
     @Override
-    public CancellableTask getProducts(final int page, final Long collectionId, final Collection.SortOrder sortOrder, final Callback<List<Product>> callback) {
-        return new CancellableTaskSubscriptionWrapper(getProducts(page, collectionId, sortOrder).subscribe(new InternalCallbackSubscriber<>(callback)));
+    public CancellableTask getProductTags(int page, Callback<List<String>> callback) {
+        return new CancellableTaskSubscriptionWrapper(getProductTags(page).subscribe(new InternalCallbackSubscriber<>(callback)));
     }
 
     @Override
-    public Observable<List<Product>> getProducts(final int page, final Long collectionId, final Collection.SortOrder sortOrder) {
+    public Observable<List<String>> getProductTags(final int page) {
         if (page < 1) {
             throw new IllegalArgumentException("page is a 1-based index, value cannot be less than 1");
         }
-        if (collectionId == null) {
-            throw new NullPointerException("collectionId cannot be null");
-        }
-        if (sortOrder == null) {
-            throw new NullPointerException("sortOrder cannot be null");
-        }
 
         return retrofitService
-            .getProducts(appId, collectionId, pageSize, page, sortOrder.toString())
+            .getProductTagPage(appId, page, pageSize)
             .retryWhen(networkRetryPolicyProvider.provide())
             .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>())
-            .compose(new UnwrapRetrofitBodyTransformer<ProductListings, List<Product>>())
-            .onErrorResumeNext(new BuyClientExceptionHandler<List<Product>>())
+            .compose(new UnwrapRetrofitBodyTransformer<ProductTagsWrapper, List<ProductTag>>())
+            .map(new Func1<List<ProductTag>, List<String>>() {
+                @Override
+                public List<String> call(final List<ProductTag> productTags) {
+                    if (productTags == null) {
+                        return null;
+                    } else {
+                        final List<String> tags = new ArrayList<>();
```

Comment:
I know it's trivial, but would be much nicer if this mapping was extracted to a separate function.